python/cudf/cudf/core/column/decimal.py

```diff
@@ -224,6 +224,13 @@ def fillna(
 
         Returns a copy with null filled.
         """
+        if isinstance(value, (int, Decimal)):
```

Comment:
Should we also allow floating types here? If not, we should probably raise an error that is clear.